### PR TITLE
fix(surveys): enable app warning and setup help fixes

### PR DESCRIPTION
**Pull request by liyiy**
## Problem

Non API surveys will literally not work unless the app is enabled and the "opt in site app" setting is present. 

Also no one reads things in a collapse, so updated the setup help info to make it more explicit.

## Changes
Detect if the survey app is not enabled and present a banner to make sure the user sets up the survey properly before they can use it.

Screen Shot 2023-07-13 at 10 47 32 AM
Screen Shot 2023-07-13 at 10 47 41 AM
Screen Shot 2023-07-13 at 10 50 57 AM

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: frontend/src/scenes/surveys/SurveyView.tsx [hunk 1]
```diff
@@ -103,6 +106,14 @@ export function SurveyView({ id }: { id: string }): JSX.Element {
                             </>
                         }
                     />
+                    {showSurveyAppWarning && (
+                        <LemonBanner type="error">
+                            Surveys requires the{' '}
+                            <a onClick={() => surveyPlugin?.id && editPlugin(surveyPlugin.id)}>survey app</a> to be
+                            enabled. Enable and save it, and make sure you have the "opt_in_site_apps" setting in your
```
Comment (liyiy) [inline frontend/src/scenes/surveys/SurveyView.tsx:113]: Does this sound too direct and bossy? @joethreepwood

**Comment by neilkakkar** [comment]
This is somewhat what I wanted to stress yday: Site-app being disabled doesn't mean the non-API surveys wouldnt' work, they'll work if you have your own custom setup that just replicates what the site-app does with custom styling (that's exactly what we'll do for our team, for example).

So, I'd recommend making this dismissible with a button confirming "I'll implement this myself" or something like that, so it doesn't confuse users who come later to create surveys but don't need to know how it shows up.


At the same time, if you think this is too niche, and would be used only by us, fine to keep as is.

File: frontend/src/scenes/surveys/SurveyView.tsx (hunk 1 above)
Comment (joethreepwood) [reply frontend/src/scenes/surveys/SurveyView.tsx:113]:
```suggestion
                            enabled. You also need to make sure you have the "opt_in_site_apps" setting in your
```
Not at all. This sounds a little simpler, however.

**Review by joethreepwood — APPROVED** [review]

**Comment by liyiy** [comment]
> This is somewhat what I wanted to stress yday: Site-app being disabled doesn't mean the non-API surveys wouldnt' work, they'll work if you have your own custom setup that just replicates what the site-app does with custom styling (that's exactly what we'll do for our team, for example).
> 
> So, I'd recommend making this dismissible with a button confirming "I'll implement this myself" or something like that, so it doesn't confuse users who come later to create surveys but don't need to know how it shows up.
> 
> At the same time, if you think this is too niche, and would be used only by us, fine to keep as is.

I'm a little confused here, this banner only shows up if you create a non API survey and don't have the survey app enabled, otherwise it won't show up?

**Comment by liyiy** [comment]
@neilkakkar I think we can make it more obvious during the survey creation phase or even beforehand that they can create their own API surveys but the banner shouldn't get in the way of that I think?

**Comment by neilkakkar** [comment]
The issue for our team specifically is this:

1. We disable the site-app because it doesn't fit in with our design.
2. Michael creates a PostHog themed survey shower which uses `getActiveMatchingSurveys`, AND, works automatically for any popup based surveys, just like the site app. It's effectively like building the site-app into our posthog code.

3. Now, Joe, say, who's unfamiliar with the implementation, wants to create a survey, he sees the warning when he shouldn't, because he can freely create popup surveys, and they will work, despite not having the app.

**Review by neilkakkar — APPROVED** [review]

**Comment by liyiy** [comment]
> The issue for our team specifically is this:
> 
> 1. We disable the site-app because it doesn't fit in with our design.
> 2. Michael creates a PostHog themed survey shower which uses `getActiveMatchingSurveys`, AND, works automatically for any popup based surveys, just like the site app. It's effectively like building the site-app into our posthog code.
> 3. Now, Joe, say, who's unfamiliar with the implementation, wants to create a survey, he sees the warning when he shouldn't, because he can freely create popup surveys, and they will work, despite not having the app.

So if Joe creates this survey that isn't under Michael's theme, isn't that okay? And if it's not, then Michael and Joe need to communicate about it, how does a dismissable banner help in this scenario?

**Comment by neilkakkar** [comment]
Yeah, that's the contention I guess: I don't think so.

When we have our appropriate themed implementation, Joe trying to enable the site-app is not what we want.
By dismissible, I meant permanently dismissible, such that if Michael says 'not to show this warning', it doesn't show this warning to Joe, which is right, because Joe shouldn't have to enable the site-app. He shouldn't even be prompted to do it, because we have our custom handling for non API based surveys.

---

Now that I think about it, this is perhaps an entirely different presentation in app, since it doesn't seem right to me to make it so easy for Joe to make mistakes.

(sorry Joe 😝 )

---

How about this, let's get this in, let's wait for the themed surveys to release, and see how Joe struggles, if at all?

**Comment by joethreepwood** [comment]
> > 3. Now, Joe, say, who's unfamiliar with the implementation, wants to create a survey, he sees the warning when he shouldn't, because he can freely create popup surveys, and they will work, despite not having the app.
> 
> So if Joe creates this survey that isn't under Michael's theme, isn't that okay? And if it's not, then Michael and Joe need to communicate about it, how does a dismissable banner help in this scenario?

OH, SO THIS IS MY FAULT NOW??? 😉

**Comment by liyiy** [comment]
In this theoretical situation I think it's still better we show the banner error bc if Joe isn't supposed to implement his own surveys he'll see it and be like oh... BIG ERROR banner, maybe I'll ask someone else what's going on since I'm not sure, instead of just wondering why his survey doesn't work with no context 🤔 🤣 

Anyway this is still pretty niche and theoretical, always happy to make changes to fix this if users encounter problems

**Comment by neilkakkar** [comment]
> instead of just wondering why his survey doesn't work with no context

That's the thing, his survey will work perfectly fine, because we copied the site app. The warning is just a misdirection in this case.

**Comment by joethreepwood** [comment]
If it helps: I'm already confused.

**Comment by annikaschmid** [comment]
Survey is type "popover" AND survey app is not enabled = show warning on the specific survey page

Survey is type "API" AND survey app is not enabled = don't show warning on the specific survey page

We can adjust the wording in the warning to say "Popover surveys require the survey app to be turned on..."

I don't understand why this wouldn't work?

If it's a company's rule that they never want to use popover surveys and always use API/their homegrown tool, I don't think we have to support this rule at this point. Way to edge-casey...

**Comment by neilkakkar** [comment]
Conscious I might be overcomplicating. I still think it's going to come up reasonably often, but I'll wait for someone other than me to raise this point before investing more time into thinking of a solution here.

So, for now, let's just say "Popup mode surveys can't be customised using code, and you have to use the site-app to use popup mode". In which case this works perfectly.

**Pull request merged**